snapcraft/internal/sources/errors.py

```diff
@@ -23,3 +23,11 @@ class IncompatibleOptionsError(errors.SnapcraftError):
 
     def __init__(self, message):
         super().__init__(message=message)
+
+
+class ChecksumDoesNotMatchError(errors.SnapcraftError):
+
+    fmt = '{message}'
+
+    def __init__(self, message):
+        super().__init__(message=message)
```

Comment:
This would do more for you if it accepted an expected and actual digest and printed a decent formatted message on its own.